### PR TITLE
Convection-diffusion solver draft for review

### PR DESCRIPTION
**Pull request by AliKarakus**
Any feedback is welcome such as  connection type to INS, separating flow side BC conditions from scalar solver, etc...

### CODE REVIEW

**Comment by noelchalmers** [comment]
Hi @AliKarakus , this looks like great work. I've confirmed this is compiling and working for tris, tets, and hexes for serial and MPI. I've noticed the things hang with >8 ranks, is that an issue in the core of the code of something about this solver? Also it appears the quads version in unstable.

I like the Gauss Hill default config. @AliKarakus , could you add an analogous config for the 3D runs as well? 

Question for the group: how do we feel about the cds name? It's nice and terse, but maybe the solver folder could have a more descriptive name?

I'll cherry pick this into my staging branch after we merge to master and I'll convert the build system to the new version, and eventually update the class structures as I'm doing in the rest of the solvers. 

Thanks again @AliKarakus

File: solvers/cds/data/cdsGaussHill2D.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
```
Comment (noelchalmers) [inline solvers/cds/data/cdsGaussHill2D.h]: Make a 3D version?

**Comment by AliKarakus** [comment]
Thanks @noelchalmers, will check the issues.

**Comment by noelchalmers** [comment]
This PR was refactored and added as the FokkerPlanck solver in 40dfb0b4bc659a94d0a21bc69ca9729540a78100

**Pull request closed without merging**